ExampleMod/Items/ExampleDashAccessory.cs

```diff
@@ -0,0 +1,208 @@
+			item.rare = ItemRarityID.Blue;
+			item.value = Item.sellPrice(silver: 60);
+		}
+
+		public override void UpdateAccessory(Player player, bool hideVisual)
+		{
+			ExampleDashPlayer mp = player.GetModPlayer<ExampleDashPlayer>();
+
+			//If the dash is active, decrement the timers and do other things
+			if(mp.DashActive)
+			{
+				//This is where we set the afterimage effect.  You can replace these two lines with whatever you want to happen during the dash
+				//Some examples include:  spawning dust where the player is, adding buffs, making the player immune, etc.
+				//Here we take advantage of "player.eocDash" and "player.armorEffectDrawShadowEOCShield" to get the Shield of Cthulhu's afterimage effect
+				player.eocDash = mp.DashTimer;
+				player.armorEffectDrawShadowEOCShield = true;
+
+				//If the dash has just started, apply the dash velocity in whatever direction we wanted to dash towards
+				if(mp.DashTimer == ExampleDashPlayer.MAX_DASH_TIMER)
+				{
+					Vector2 newVelocity = player.velocity;
+					
+					//Only apply the dash velocity if our current speed in the wanted direction is less than DashVelocity
+					if((mp.PressedUp && player.velocity.Y > -mp.DashVelocity) || (mp.PressedDown && player.velocity.Y < mp.DashVelocity))
+					{
+						//Y-velocity is set here
+						//If the direction requested was Up, then we adjust the velocity to make the dash appear "faster" due to gravity being immediately in effect
+						//This adjustment is roughly 1.3x the intended dash velocity
+						float dashDirection = mp.PressedDown ? 1 : -1.3f;
+						newVelocity.Y = dashDirection * mp.DashVelocity;
+					}
+					else if((mp.PressedLeft && player.velocity.X > -mp.DashVelocity) || (mp.PressedRight && player.velocity.X < mp.DashVelocity))
+					{
+						//X-velocity is set here
+						int dashDirection = mp.PressedRight ? 1 : -1;
+						newVelocity.X = dashDirection * mp.DashVelocity;
+					}
+
+					player.velocity = newVelocity;
+				}
+
+				mp.DashTimer--;
+				mp.DashDelay--;
+
+				if(mp.DashDelay == 0)
+				{
+					//The dash has ended.  Reset the fields
+					mp.DashDelay = ExampleDashPlayer.MAX_DASH_DELAY;
+					mp.DashTimer = ExampleDashPlayer.MAX_DASH_TIMER;
+					mp.DashActive = false;
+				}
+			}
+		}
+
+		public override void AddRecipes()
+		{
+			ModRecipe recipe = new ModRecipe(mod);
+			recipe.AddIngredient(ItemType<ExampleItem>(), 10);
+			recipe.AddTile(TileType<ExampleWorkbench>());
+			recipe.SetResult(this);
+			recipe.AddRecipe();
+		}
+	}
+
+	public class ExampleDashPlayer : ModPlayer
+	{
+		//This array will be used to keep track of the previous values of "player.doubleTapCardinalTimer"
+		private int[] oldCardinalTimers = new int[4];
+
+		//These indicate what direction is what in the timer arrays used
+		private readonly int Down = 0;
+		private readonly int Up = 1;
+		private readonly int Right = 2;
+		private readonly int Left = 3;
+
+		//Bools for if a direction was pressed in any of the cardinal directions
```

Comment:
You could use an array here, it'd be nicer to see you use Up/Down/Right/Left for all `oldCardinalTimers` `Pressed` and `DoubleTap`